### ISSUE TITLE
Basic Auth ClassCastException

### ISSUE THREAD

**Issue by jsap2018** [pinpoint-apm/pinpoint#8016]
#### What version of pinpoint are you using?
master

#### Describe your problem**
I enabled basic authentication feature mentioned here: https://github.com/pinpoint-apm/pinpoint/issues/7205.

Navigating to Pinpoint brings up the login page, however, once logged in I receive the following error.

How do I fix this issue?

![image](https://user-images.githubusercontent.com/39391821/124998036-503a5c80-e011-11eb-8b98-0041fd706fec.png)

Logs:

```
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]: java.lang.ClassCastException: org.springframework.security.core.userdetails.User cannot be cast to java.lang.String
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.service.UserServiceImpl.getUserIdFromSecurity(UserServiceImpl.java:137) ~[classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.service.UserServiceImpl$$FastClassBySpringCGLIB$$6b718a25.invoke(<generated>) ~[classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) ~[spring-tx-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388) ~[spring-tx-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.service.UserServiceImpl$$EnhancerBySpringCGLIB$$3b43debf.getUserIdFromSecurity(<generated>) ~[classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.controller.ConfigController.getProperties(ConfigController.java:66) ~[classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.controller.ConfigController$$FastClassBySpringCGLIB$$d173fc84.invoke(<generated>) ~[classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89) ~[spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.interceptor.PerformanceLoggingInterceptor.logging(PerformanceLoggingInterceptor.java:26) [classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at sun.reflect.GeneratedMethodAccessor66.invoke(Unknown Source) ~[?:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692) [spring-aop-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.controller.ConfigController$$EnhancerBySpringCGLIB$$b65e515d.getProperties(<generated>) [classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_291]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197) [spring-web-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141) [spring-web-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106) [spring-webmvc-5.3.6.jar!>
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894) [spring-webmvc-5.3.6.ja>
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808) [spring-webmvc-5.3.6.jar!/:5>
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1060) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:962) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at javax.servlet.http.HttpServlet.service(HttpServlet.java:645) [javax.servlet-api-4.0.1.jar!/:4.0.1]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883) [spring-webmvc-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at javax.servlet.http.HttpServlet.service(HttpServlet.java:750) [javax.servlet-api-4.0.1.jar!/:4.0.1]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) [tomcat-embed-websocket-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at com.navercorp.pinpoint.web.servlet.RewriteForV2Filter.doFilter(RewriteForV2Filter.java:92) [classes!/:2.3.0-SNAPSHOT]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [tomcat-embed-core-9.0.45.jar!/:?]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.filter.ShallowEtagHeaderFilter.doFilterInternal(ShallowEtagHeaderFilter.java:106) [spring-web-5.3.6.jar!/:5.3.6]
Jul 08 22:21:45 10-222-86-182.ssnc-corp.cloud java[112371]:         at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) [spring-web-5.3.6.jar!/:5.3.6]

```

**Comment by jsap2018** [comment]
@koo-taejin 

Have you ran into this issue before?

**Comment by koo-taejin** [comment]
hi @jsap2018 

I will see if it has something wrong . 

thanks :)

**Comment by koo-taejin** [comment]
@jsap2018 

I have solved this issue via https://github.com/pinpoint-apm/pinpoint/pull/8017

Please check whether it works well or not. 

Thanks :)

**Comment by jsap2018** [comment]
> 
> 
> @jsap2018
> 
> I have solved this issue via #8017
> 
> Please check whether it works well or not.
> 
> Thanks :)

Works now, thanks!
